app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt

```diff
@@ -1,9 +1,7 @@
 package nl.eduvpn.app.entity
 
-import com.wireguard.config.Config
-
 sealed class VpnProtocol
```

Comment:
I think you can rewrite this like:

```
sealed class VpnProtocol {
  data class OpenVPN(val profile: Profile) : VpnProtocol()
  object WireGuard : VpnProtocol()
}
```
And later you can refer to this like:
VpnProtocol.OpenVpn(profile) or VpnProtocol.WireGuard

WireGuard.INSTANCE looks a bit weird to me

Reply:
> WireGuard.INSTANCE looks a bit weird to me

I was curious how this worked, so I compiled the apk and then decompiled the apk using dex2jar and jadx to retrieve the Java code of the WireGuard class:

```java
public static final class WireGuard extends VpnProtocol {
    public static final WireGuard INSTANCE = new WireGuard();

    private WireGuard() {
        super(null);
    }
}
```

Using WireGaurd.INSTANCE seems to be the only way to get access to the WireGuard object from Java.

Reply:
Ah yes that is still Java, did not see that. Then it is OK as it is.